internal/authentication/spnego_keytab_provider.go

```diff
@@ -0,0 +1,56 @@
+package authentication
+
+import (
+	"sync"
+
+	"github.com/authelia/authelia/v4/internal/configuration/schema"
+	"github.com/jcmturner/gokrb5/v8/keytab"
+)
+
+type SPNEGOKeytabProvider struct {
+	config *schema.SPNEGO
+	kt     *keytab.Keytab
+	mutex  sync.Mutex
+}
+
+func NewSPNEGOKeytabProvider(config *schema.SPNEGO) (*SPNEGOKeytabProvider, error) {
+	keyTabFile := "/etc/krb5.keytab"
+	if config.Keytab != "" {
+		keyTabFile = config.Keytab
+	}
```

Comment:
```suggestion
```

We wouldn't configure a default keytab. Users will have to explicitly configure it.